### PR TITLE
cart-wishlist-management

### PR DESCRIPTION
**Pull request by BelideNikhil**
Features

+ cart management
+ Wishlist management
+ cart card
+ cart summary
+ Wishlist cart

Preview: https://ninty-seven-x-store.netlify.app/

### CODE REVIEW

File: src/Components/CartCard/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                <div className="cart-qty-wrapper flex-row-center-center">
+                    <button
+                        className="btn-icon btn-icon-sm"
+                        disabled={cartLoading}
+                        onClick={() => (qty === 1 ? deleteFromCartHandler(_id) : updateQtyHandler(_id, "decrement"))}
+                    >
+                        <i className="fas fa-minus"></i>
+                    </button>
+                    <strong className="ma-8">{qty}</strong>
+                    <button
```
Comment (poornima-alugubelly) [inline src/Components/CartCard/CartCard.jsx:35]: Make this a b tag for better accessibility

File: src/Components/CartSummary/CartSummary.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
+.cart-summary {
+    width: 100%;
+    max-width: 34rem;
+    border: 1px solid var(--clr-primary-faded);
+    border-radius: var(--border-radius);
+    padding: var(--padding-md);
```
Comment (msdeshmukh009) [inline src/Components/CartSummary/CartSummary.css:6]: You can use utilities here.

File: src/Components/CartSummary/CartSummary.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        cartList.reduce((total, current) => total + Number(current.price * current.qty), 0)
+    );
+    const actualPrice = Math.round(
+        cartList.reduce(
+            (total, current) =>
+                total + Number((current.price + (current.discountPercent / 100) * current.price) * current.qty),
+            0
+        )
+    );
+    const gstTotal = Math.round((discountedPrice / 100) * 18);
```
Comment (msdeshmukh009) [inline src/Components/CartSummary/CartSummary.jsx:17]: You can reduce these functions into a single function.

File: src/Context/AuthContext.js [hunk 4]
```diff
@@ -1,67 +1,87 @@
...
+                loginError: "",
+                signupError: "",
+                token: payload.token,
+            };
         case "SET_LOGIN_ERROR":
             return { ...authState, isAuth: payload.auth, loginError: payload.error };
         case "SET_SIGNUP_ERROR":
             return { ...authState, isAuth: payload.auth, signupError: payload.error };
+        case "SET_AUTH_RENDER":
+            console.log(payload);
```
Comment (msdeshmukh009) [inline src/Context/AuthContext.js]: Remove Console

**Review by poornima-alugubelly — APPROVED** [review]

**Review by msdeshmukh009 — APPROVED** [review]
The code is readable and also UI is looking good.

File: src/Components/CartCard/CartCard.jsx (hunk 1 above)
Comment (BelideNikhil) [reply src/Components/CartCard/CartCard.jsx:35]: done!

File: src/Context/AuthContext.js (hunk 4 above)
Comment (BelideNikhil) [reply src/Context/AuthContext.js]: done!

File: src/Components/NavAside/NavAside.jsx [hunk 5]
```diff
@@ -1,6 +1,15 @@
...
+import { useAuth } from "../../Context/AuthContext";
 export default function NavAside({ navToggle, asideToggleFunction, setNavToggle }) {
+    const {
+        authState: { isAuth },
+    } = useAuth();
+    const navigate = useNavigate();
+    function asideNavigate(navigateTo) {
+        navigate(navigateTo);
+        asideToggleFunction();
+    }
```
Comment (sakshamAK) [inline src/Components/NavAside/NavAside.jsx:12]: I would suggest you to give a new line after a function ends, or while starting a new function as it looks much cleaner to read
like
```
const a = 5;

const func = () => {
}

const b = 5 
```

File: src/Context/WishlistContext.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import { useAuth } from "./AuthContext";
+import { addToWishlistService, deleteWishlistService, getWishlistService } from "../Services";
+
+const WishlistContext = createContext();
+
+function wishlistReducer(wishlistState, { type, payload }) {
+    switch (type) {
+        case "WISHLIST_RENDER":
+            return { ...wishlistState, wishList: payload.wishList };
+        case "UPDATE_WISHLIST":
```
Comment (sakshamAK) [inline src/Context/WishlistContext.js:12]: I would suggest you use variables as action types.

**Review by sakshamAK — APPROVED** [review]
Great work! You've written a good code. I loved how you created a separate file for routes and used barrel exports in services

**Pull request merged**